control-userlayer/src/main/java/org/oskari/control/userlayer/CreateUserLayerHandler.java

```diff
@@ -1,222 +1,250 @@
-import org.oskari.map.userlayer.domain.GPXGeoJsonCollection;
-import org.oskari.map.userlayer.domain.KMLGeoJsonCollection;
-import org.oskari.map.userlayer.domain.MIFGeoJsonCollection;
-import org.oskari.map.userlayer.domain.SHPGeoJsonCollection;
-import org.oskari.map.userlayer.service.GeoJsonWorker;
-import org.oskari.map.userlayer.service.UserLayerDataService;
+import fi.nls.oskari.service.ServiceException;
 import fi.nls.oskari.util.IOHelper;
 import fi.nls.oskari.util.JSONHelper;
 import fi.nls.oskari.util.PropertyUtil;
-import org.apache.commons.fileupload.FileItem;
-import org.apache.commons.fileupload.FileUploadException;
-import org.apache.commons.fileupload.disk.DiskFileItemFactory;
-import org.apache.commons.fileupload.servlet.ServletFileUpload;
-import org.json.JSONObject;
-
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import java.io.File;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.io.UnsupportedEncodingException;
-import java.util.*;
-import java.util.zip.ZipEntry;
-import java.util.zip.ZipInputStream;
 
 /**
- * Store zipped file set to oskari_user_store database
+ * CreateUserLayer allows users to upload a collection of features to be stored in the system.
+ *
+ * The uploaded file must be a ZIP file and must contain a set of valid files within.
+ * Currently supported file formats are:
+ * - GPX ({name}.gpx file must be found within the zip)
+ * - KML ({name}.kml)
+ * - MIF ({name}.mif)
+ * - SHP ({name}.shp)
+ * @see org.oskari.map.userlayer.input.FeatureCollectionParsers
+ *
+ * For some of the file formats (GPX, KML) the coordinate reference system is fixed
+ * (both use WGS84 lon,lat coordinates). For MIF and SHP we try to detect the coordinate
+ * reference system automatically. If the detection fails (for example there's no .prj file
+ * in the SHP case) we use client submitted value ('sourceEpsg' parameter) as a fallback.
  */
 @OskariActionRoute("CreateUserLayer")
 public class CreateUserLayerHandler extends ActionHandler {
 
-    private static final Logger log = LogFactory
-            .getLogger(CreateUserLayerHandler.class);
-    private static final List<String> ACCEPTED_FORMATS = Arrays.asList("SHP", "KML", "GPX", "MIF");
-    private static final String IMPORT_SHP = ".SHP";
-    private static final String IMPORT_GPX = ".GPX";
-    private static final String IMPORT_MIF = ".MIF";
-    private static final String IMPORT_KML = ".KML";
+    private static final Logger log = LogFactory.getLogger(CreateUserLayerHandler.class);
     private static final String PARAM_EPSG_KEY = "epsg";
     private static final String PARAM_SOURCE_EPSG_KEY = "sourceEpsg";
     private static final String USERLAYER_MAX_FILE_SIZE_MB = "userlayer.max.filesize.mb";
-    private static final String USERLAYER_DEFAULT_SOURCE_EPSG = "userlayer.default.source.epsg";
+    private static final String USERLAYER_DEFAULT_TARGET_EPSG = "userlayer.default.target.epsg";
```

Comment:
Change in the properties should be mentioned in MigrationGuide.md

Reply:
Added note to migration guide about properties